examples/big_models_with_accelerate/README.md

```diff
@@ -0,0 +1,80 @@
+- Offloading parameters to CPU
+- Sharding models across multiple GPUs with pipeline-parallelism
+
+
+### Using `device_map`
+
+To enable `accelerate` features with `llmcompressor`, simple insert `device_map` in `from_pretrained` during model load.
+
+```python
+from llmcompressor.transformers import SparseAutoModelForCausalLM
+MODEL_ID = "meta-llama/Meta-Llama-3-70B-Instruct"
+
+# device_map="auto" triggers usage of accelerate
+# if > 1 GPU, the model will be sharded across the GPUs
+# if not enough GPU memory to fit the model, parameters are offloaded to the CPU
+model = SparseAutoModelForCausalLM.from_pretrained(
+    MODEL_ID, device_map="auto", torch_dtype="auto")
+```
+
+`llmcompressor` is designed to respect the `device_map`, so calls to `oneshot` will work properly.
+
+### Practical Advice
+
+When working with `accelerate`, it is important to keep in mind that CPU offloading and naive pipeline-parallelism will slow down forward passes through the model. As a result, we need to take care to ensure that the quantization methods used fit well with the offloading scheme as methods that require many forward passes though the model will be slowed down.
+
+General rules of thumb:
+- CPU offloading should only be with data-free quantization methods (e.g. PTQ with `FP8_DYNAMIC`)
+- Multi-GPU is fast enough to be used with calibration data-based methods with `sequential_update=False`
+- It is possible to use Multi-GPU with `sequential_update=True`, but the runtime will be slower
+
+## Examples
+
+We will show working examples for each use case:
+- **CPU Offloading**: Quantize `Llama-70B` to `FP8` using `PTQ` with a single GPU
+- **Multi-GPU**: Quantize `Llama-70B` to `INT8` using `GPTQ` and `SmoothQuant` with 8 GPUs
+
+### Installation
+
+Install `llmcompressor`:
+
+```bash
+pip install llmcompressor==0.1.0
+```
+
+### CPU Offloading: `FP8` Quantization with `PTQ`
+
+CPU offloading is slow. As a result, we recommend using this feature only with data-free quantization methods. For example, when quantizing a model to `fp8`, we typically use simple `PTQ` to statically quantize the weights and use dynamic quantization for the activations. These methods do not require calibration data.
+
+- `cpu_offloading_fp8.py` demonstrates quantizing the weights and activations of `Llama-70B` to `fp8` on a single GPU:
+
+```bash
+export CUDA_VISIBLE_DEVICES=0
+python cpu_offloading_fp8.py
+```
+
+The resulting model `./Meta-Llama-3-70B-Instruct-FP8-Dynamic` is ready to run with `vllm`!
+
+### Multi-GPU: `INT8` Quantization with `GPTQ`
+
+For quantization methods that require calibration data (e.g. `GPTQ`), CPU offloading is too slow. For these methods, `llmcompressor` can use `accelerate` multi-GPU to quantize models that are larger than a single GPU. For example, when quantizing a model to `int8`, we typically use `GPTQ` to statically quantize the weights, which requires calibration data.
```

Comment:
Worth mentioning that GPTQ needs to allocate extra memory, ~2x the size of the fp16 model for sequential_update=False

Reply:
thats a nice rule of thumb

btw - are the hessians the same shape as the weight matrix? I thought they were the shape of X?

Reply:
Theyre (column x column) for each linear layer, and must be fp32. So ~2x the model size extra memory is a rough rule of thumb. The exact calculation is here: https://github.com/vllm-project/llm-compressor/blob/d232ea8b3ca30a70890feac1b56f5720c6e6b3c0/src/llmcompressor/transformers/compression/helpers.py#L100